File: session6-addnlds/Calc.py
# Python does not support method overloading. We can use default arguments to achieve the same.
class Calc:
    def add(self, a=0, b=0):
        return a + b


calc = Calc()
print(calc.add())  # 0
print(calc.add(5))  # 5
print(calc.add(5, 10))  # 15
print(calc.add(5.6, 7.9))  # 13.5

File: session6-addnlds/Calc.js
// JavaScript does not support Method Overloading.
// But we can achieve it by using default values for the parameters.
class Calc{
    add(a=0, b=0) {
        return a + b;
    }
}

let calc = new Calc();
console.log(calc.add()); // 0
console.log(calc.add(10)); // 10
console.log(calc.add(10, 20)); // 30
console.log(calc.add(5.4, 3.24)); // 8.64